dex/src/main/scala/com/wavesplatform/dex/db/leveldb/Key.scala

```diff
@@ -10,8 +10,10 @@ trait Key[V] {
 
   override lazy val toString: String = BigInt(keyBytes).toString(16)
 
+  def canEqual(other: Any): Boolean = other.isInstanceOf[Key[V]]
+
   override def equals(obj: scala.Any): Boolean = obj match {
-    case that: Key[V] => java.util.Arrays.equals(this.keyBytes, that.keyBytes)
+    case that: Key[V] => that.canEqual(this) && java.util.Arrays.equals(this.keyBytes, that.keyBytes) && this.name.equals(that.name)
```

Comment:
`this.name.equals(that.name)` must be just `name == that.name`, because:
```scala
scala> val x: String = null
val x: String = null

scala> x.equals("foo")
java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "x()" is null
  ... 32 elided

scala> null == "foo"
val res1: Boolean = false
```

Reply:
fixed